commands/channel.go

```diff
@@ -263,6 +265,48 @@ func archiveChannelsCmdF(c client.Client, cmd *cobra.Command, args []string) err
 	return nil
 }
 
+func getAllPublicChannelsForTeam(c client.Client, teamID string) ([]*model.Channel, error) {
+	channels := []*model.Channel{}
+	page := 0
+
+	for {
+		channelsPage, response := c.GetPublicChannelsForTeam(teamID, page, web.PerPageMaximum, "")
+		if response.Error != nil {
+			return nil, response.Error
+		}
+
+		if len(channelsPage) == 0 {
```

Comment:
Probably not needed, but should we check later also `len(channelsPage) < web.PerPageMaximum` to save us from one more call?

Reply:
We've found instances in which although the page comes with less than the total amount of elements, there are more pages to fetch (for example, when the items are indexed on ES/bleve) hence we stop only when the page fetched comes empty, to be 100% sure that there are no more elements to fetch